CONTRIBUTING.md

```diff
@@ -1,221 +1,295 @@
+more can be found in the `/tests` directory. However, we also test all of our desktop examples as well. 
 
-* add_graphic - add a `Graphic` to the `PlotArea`, append to the end of the `PlotArea._graphics` list
-* insert_graphic - insert a `Graphic` to the `PlotArea`, insert to a specific position of the `PlotArea._graphics` list
-* remove_graphic - remove a graphic from the `Scene`, **does not delete it**
-* delete_graphic - delete a graphic from the `PlotArea`, performs garbage collection
-* clear - deletes all graphics from the `PlotArea`
-* center_graphic - center camera w.r.t. a `Graphic`
-* center_scene - center camera w.r.t. entire `Scene`
-* auto_scale - Auto-scale the camera w.r.t to the `Scene`
+If you are adding a new example to the library, you will need to add to comments to the top of your `.py` file in order to make sure it is both tested and added to the gallery. 
 
-In addition, `PlotArea` supports `__getitem__`, so you can do: `plot_area["graphic_name"]` to retrieve a `Graphic` by 
-name :smile: 
+```python
+# test_example = true
+# sphinx_gallery_pygfx_docs = 'screenshot'
+```
 
-You can also check if a `PlotArea` has certain graphics, ex: `"some_image_name" in plot_area`, or `graphic_instance in plot_area`
+### Documentation
 
-#### Subplot
+Documentation is a crucial part of open-source software and greatly influences the ability to use a codebase. As such, it is imperative that any new changes are
+properly documented as outlined below. 
 
-This class inherits from `PlotArea` and `GraphicMethodsMixin`.
+We use [`sphinx`](https://www.sphinx-doc.org/en/master/) for generating our documentation. In addition to this, we also use the [`sphinx-gallery`](https://sphinx-gallery.github.io/stable/index.html)
+extension to build our examples gallery.
 
-`GraphicMethodsMixin` is a simple class that just has all the `add_<graphic>` methods. It is autogenerated by a utility script like this:
+If you would like to build the documentation locally:
 
 ```bash
-python scripts/generate_add_methods.py
-```
+cd docs
+# regenerate the api guide
+python source/generate_api.py
 
-Each `add_<graphic>` method basically creates an instance of `Graphic`, adds it to the `Subplot`, and returns a weakref 
-proxy to the `Graphic`.
+# build locally
+make html
+```
 
-Subplot has one property that is not in `PlotArea`:
+#### Adding documentation
 
-* docks: a `dict` of `PlotAreas` which are located at the "top", "right", "left", and "bottom" edges of a `Subplot`. By default their size is `0`. They are useful for putting things like histogram LUT tools.
+All public-facing functions and classes should have complete docstrings, which start with a one-line short summary of the function, 
+a medium-length description of the function / class and what it does, and a complete description of all arguments and return values. 
+Docstrings should be relatively short, providing the information necessary for a user to use the code.
```

Comment:
```suggestion
Docstrings should comprehensive, providing the information necessary for a user to use the method or property without going through the code.
```

Our docstrings are  not short XD, good libs generally have comprehensive docstrings, see sklearn.